File: src/main/java/com/theonlytails/ruby/registries/EnchantmentRegistry.java
package com.theonlytails.ruby.registries;

import com.theonlytails.ruby.RubyMod;
import com.theonlytails.ruby.enchantments.StingerEnchantment;
import net.minecraft.enchantment.Enchantment;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class EnchantmentRegistry {
    public static final DeferredRegister<Enchantment> ENCHANTMENTS = DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, RubyMod.MOD_ID);

    public static final RegistryObject<Enchantment> STINGER = ENCHANTMENTS.register("stinger", StingerEnchantment::new);
}

File: src/main/kotlin/com/theonlytails/rubymod/registries/EnchantmentRegistry.kt
@file:Suppress("unused")

package com.theonlytails.rubymod.registries

import com.theonlytails.rubymod.MOD_ID
import com.theonlytails.rubymod.enchantments.StingerEnchantment
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Registers custom enchantments.
 *
 * @author TheOnlyTails
 */
object EnchantmentRegistry {
	val enchantments = KDeferredRegister(ForgeRegistries.ENCHANTMENTS, MOD_ID)

	val stinger by enchantments.registerObject("stinger", ::StingerEnchantment)
}
